### ISSUE TITLE
Sorting should work for array scalars.

### ISSUE THREAD

**Issue by charris** [numpy/numpy#7051]
Just return the array, don't error.
```In [10]: a = ones(())

In [11]: a.ndim
Out[11]: 0

In [12]: a.sort()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-12-e7eb8b51a6fa> in <module>()
----> 1 a.sort()

ValueError: axis(=-1) out of bounds
```

**Comment by tushar-rishav** [comment]
@charris Checking ` a.ndim ` for zero before calling ` a.sort ` [here] (https://github.com/numpy/numpy/blob/master/numpy/core/fromnumeric.py#L825) avoids the ValueError while computing ` np.sort(a) ` for array scalars.  I've this tried this and it worked
```py
if not a.ndim:
    a = asanyarray([a])
a.sort(axis, kind, order)
return a
```
Probably, doing just that won't help because we'll still get ValueError in case of ` a.sort() `.  I noticed that behind the curtain `a.sort()` calls `PyArray_Sort` which takes `PyArrayObject` instance , `axis` and `order` . Unfortunately, I couldn't get much insight about how `ndarray.sort()` has been implemented here. 

```cpp
#define PyArray_Sort  (*(int (*)(PyArrayObject *, int, NPY_SORTKIND))  PyArray_API[129])
```

As I am a beginner I might be missing something here. Could you please let me know where `PyArrayObject` has been declared?
Thanks in advance!

**Comment by charris** [comment]
The base sort is an array method, so that is what we want fixed.

**Comment by seberg** [comment]
I agree that it is silly that this exists and does not work, but on the other hand the default for the sort is axis=-1, which is correctly an error for 0-d arrays unlike most similar methods which use axis=None (unsupported currently in sort) where this is no problem.
I would be +1 to allow `arr.sort(None)`, which would also make sense for 0-d. But I don't feel just removing the error is more canonical, so I am slightly against it unless there is at least some use case.

**Comment by charris** [comment]
I ran into this because of an argument that could be either a list or a scalar and I wanted to preserve that distinction while also sorting the result of asarray. So in that case it would avoid special casing.

**Comment by jaimefrio** [comment]
Do we want to fix this for the Python API only, or for `PyArray_Sort` as well? Should array scalars suceed regardless of the value of `axis`, or should they only do for `axis=-1`?

On a related note, is this behavior correct, or should the return also be an array scalar?

```Python
>>> np.argsort(np.array(8))
array([0])
```

**Comment by njsmith** [comment]
I'm not sure that this does make sense... sorting needs a dimension to
operate on, doesn't it? You could flatten, but sort is inplace so that
doesn't make much sense either...

**Comment by charris** [comment]
@njsmith Array scalars can be operated on inplace. I guess I'm looking at 0-D arrays as containers rather than scalars even if they can't be indexed with integers.

**Comment by charris** [comment]
Mind, i'd just like the operation to be a NOP rather than an error.

**Comment by njsmith** [comment]
I'm just saying that sorting is an intrinsically one-dimensional operation
(it feels like to me). So I can see an argument for zerod_array.sort()
being equivalent to zerod_array.resize((1,)) (with the sort implicitly
doing an in-place flatten to get something one-dimensional and sortable,
and then sorting it, which would be a no-op because there's only one
element). But zerod_array.sort() doesn't make sense to me as a no-op that
leaves you with a zero-d output.
